pkgs/development/python-modules/pycategories/default.nix

```diff
@@ -0,0 +1,32 @@
+{ buildPythonPackage
+, pytestcov
+, fetchPypi
+, lib
+, pytest
+, pythonOlder
+, pytestrunner
+}:
+
+buildPythonPackage rec {
+  pname = "pycategories";
+  version = "1.1.0";
+  disabled = pythonOlder "3.4";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1vkza285jr2qzzq6bnyas93xx6my8rn3smc3vsdwa6kxx660l96w";
+  };
+
+  nativeBuildInputs = [ pytestrunner ];
+  buildInputs = [ (import ./infix.nix { inherit buildPythonPackage lib fetchPypi; }) ];
+  checkInputs = [ pytest pytestcov ];
+  # Python packages don't have a checkPhase, only an installCheckPhase
```

Comment:
`checkPhase` is actually used as the `installCheckPhase` in python packages, so this is in fact disabling the execution of any tests present.

Reply:
The problem you're having with the tests is because this author doesn't include the tests in the pypi package. In situations like this you can instead grab the source straight from the repo, in this case:

```
  src = fetchFromGitLab {
    owner = "danielhones";
    repo = pname;
    rev = "v${version}";
    sha256 = "1payrvy3diiv05dvwjf17vrf1ipl1ddx17fanksc5xd221l9qggl";
  };
```

remove the `doCheck = false;` and it all works for me.

Reply:
I may be able to fix it upstream. How would one include the tests in the pypi package? I'm having no luck with finding documentation on how to do this.

Reply:
That's a good question. It might involve including it in a manifest? I'm usually satisfied with pulling the source from github.

Reply:
@dmvianna I just made a PR about this for the `msrestazure` repo, you can take a look. It will include tests in the source package, but not wheels. https://github.com/Azure/msrestazure-for-python/pull/133

Reply:
since you have a top-level tests directory, this should be exactly what you're looking for